jwql/instrument_monitors/common_monitors/readnoise_monitor.py

```diff
@@ -0,0 +1,625 @@
+        plt.title('{}'.format(outname))
+
+        # Save the figure
+        plt.savefig(output_filename, bbox_inches='tight', dpi=200, overwrite=True)
+        set_permissions(output_filename)
+        logging.info('\t{} created'.format(output_filename))
+
+        return output_filename
+
+    def make_crds_parameter_dict(self):
+        """Construct a paramter dictionary to be used for querying CRDS
+        for the current reffiles in use by the JWST pipeline.
+
+        Returns
+        -------
+        parameters : dict
+            Dictionary of parameters, in the format expected by CRDS.
+        """
+
+        parameters = {}
+        parameters['INSTRUME'] = self.instrument.upper()
+        parameters['DETECTOR'] = self.detector.upper()
+        parameters['READPATT'] = self.read_pattern.upper()
+        parameters['SUBARRAY'] = self.subarray.upper()
+        parameters['DATE-OBS'] = datetime.date.today().isoformat()
+        current_date = datetime.datetime.now()
+        parameters['TIME-OBS'] = current_date.time().isoformat()
+
+        return parameters
+
+    def make_histogram(self, data):
+        """Creates a histogram of the input data and returns the bin centers 
+        and the counts in each bin.
+
+        Parameters
+        ----------
+        data : numpy.ndarray
+            The input data.
+
+        Returns
+        -------
+        n : numpy.ndarray
+            The counts in each histogram bin.
+
+        bin_centers : numpy.ndarray
+            The histogram bin centers.
+        """
+
+        # Calculate the histogram range as that within 5 sigma from the mean
+        data = data.flatten()
+        clipped = sigma_clip(data, sigma=3.0, maxiters=5)
+        mean, stddev = np.nanmean(clipped), np.nanstd(clipped)
+        lower_thresh, upper_thresh = mean - 4 * stddev, mean + 4 * stddev
+
+        # Some images, e.g. readnoise images, will never have values below zero
+        if (lower_thresh < 0) & (len(data[data < 0]) == 0):
+            lower_thresh = 0.0
+
+        # Make the histogram
+        n, bin_edges = np.histogram(data, bins='auto', range=(lower_thresh, upper_thresh))
+        bin_centers = (bin_edges[:-1] + bin_edges[1:]) / 2
+
+        return n, bin_centers
+
+    def make_readnoise_image(self, data):
+        """Calculates the readnoise for the given input dark current ramp.
+
+        Parameters
+        ----------
+        data : numpy.ndarray
+            The input ramp data. The data shape is assumed to be a 4D array in
+            DMS format (integration, group, y, x).
+
+        Returns
+        -------
+        readnoise : numpy.ndarray
+            The 2D readnoise image.
+        """
+
+        # Create a stack of CDS images (group difference images) using the input ramp data,
+        # combining multiple integrations if necessary.
+        logging.info('\tCreating stack of CDS difference frames')
+        n_ints, n_groups, n_y, n_x = data.shape
```

Comment:
I would suggest to use slightly more explicit variable names here, maybe something like `num_ints`, `num_groups`.  It is small, but could help with the readability.